compiler/resolution/functionResolution.cpp

```diff
@@ -2719,12 +2719,60 @@ static bool typeUsesForwarding(Type* t);
 
 static bool populateForwardingMethods(CallInfo& info);
 
+void trimVisibleCandidates(CallInfo&       info,
+                           Vec<FnSymbol*>& mostApplicable,
+                           Vec<FnSymbol*>& visibleFns) {
+  CallExpr* call = info.call;
+
+  bool isInit = false;
+  if (call->numActuals() >= 2 && call->isNamed("init")) {
+    if (SymExpr* se = toSymExpr(call->get(1))) {
+      isInit = se->symbol() == gMethodToken;
+    }
+  }
+
+  bool isNew = call->numActuals() >= 1 && call->isNamed("_new");
```

Comment:
I wouldn't normally stress about call->isNamed but at this point in the compiler it might be worth avoiding a strcmp  and having e.g. isNamedAstr.